src/Simulation/qdk_sim_rs/docs/c-api.md

```diff
@@ -0,0 +1,147 @@
+# Using Experimental Simulators from C
+
+This module exposes a C API for this crate, useful for embedding into simulation
+runtimes.
+
+## Safety
+
+As this is a foreign-function interface, many of the functions exposed here
+are **unsafe**, representing that the caller is required to ensure that safety
+conditions in the host language are upheld.
+
+Please pay attention to any listed safety notes when calling into this C
+API.
+
+## Generating and Using C API Headers
+
+The [`qdk_sim`](..) crate has enabled the use of [`cbindgen`](https://crates.io/crates/cbindgen), such that C-language header files are generated automatically as part of the build for this crate.
+
+```bash
+cargo install --force cbindgen
+cbindgen --language C --output include/qdk_sim.h
+cbindgen --language C++ --output include/qdk_sim.hpp
+```
+
+This will generate `include/qdk_sim.h` and `include/qdk_sim.hpp, which can then be used from C and C++ callers, respectively. For example, to call from C:
+
+```c
+#include <stdio.h>
+#include "qdk_sim.h"
+
+int main() {
+    uintptr_t sim_id;
+    uintptr_t result0, result1;
+
+    init(2, "mixed", &sim_id);
+    h(sim_id, 0);
+    h(sim_id, 1);
+
+    m(sim_id, 0, &result0);
+    m(sim_id, 1, &result1);
+
+    printf("got %llu %llu", result0, result1);
+    destroy(sim_id);
+}
+```
+
+To build and run the above example using Clang on Windows:
+
+```bash
+$ clang example.c -Iinclude -Ltarget/debug -lqdk_sim -lws2_32 -lAdvapi32 -lUserenv
+got 1 1
+```
```

Comment:
Nit: The run command is skipped (I see the build command. Then the output of the run command is shown).

Reply:
Good point, thanks for catching that!

```suggestion
```bash
$ clang example.c -Iinclude -Ltarget/debug -lqdk_sim -lws2_32 -lAdvapi32 -lUserenv
$ ./a.exe
got 1 1
```
```